app/main/routes.py

```diff
@@ -733,48 +731,32 @@ def download_record(record_id: uuid.UUID):
 
     validate_body_user_groups_or_404(file.consignment.series.body.Name)
 
-    s3 = boto3.client("s3")
     bucket = current_app.config["RECORD_BUCKET_NAME"]
     key = f"{file.consignment.ConsignmentReference}/{file.FileId}"
 
     try:
-        s3_file_object = s3.get_object(Bucket=bucket, Key=key)
-    except Exception as e:
-        current_app.app_logger.error(f"Failed to get object from S3: {e}")
-        abort(404)
-
-    download_filename = file.FileName
+        s3.head_object(Bucket=bucket, Key=key)
+    except s3.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] == "404":
```

Comment:
currently this will continue without raising an error if the error from s3 is not a 404. Maybe we should raise any other error as a 500 unexpected error?